zebra/rt_netlink.c

```diff
@@ -3695,14 +3695,6 @@ static ssize_t netlink_neigh_update_ctx(const struct zebra_dplane_ctx *ctx,
 		/* local neigh */
 		if (update_flags & DPLANE_NEIGH_SET_STATIC)
 			ext_flags |= NTF_E_MH_PEER_SYNC;
-
```

Comment:
The commit comment says that the WEAK_OVERRIDE flag is preventing us setting a remote NEIGH as stale when it transitions to be a local NEIGH. But this chunk seems to indicate that the flag is only set on local NEIGHs. What does the sequence of events look like here?

Reply:
That is correct, the VM move is from remote (rack-2, tor-21/tor-22) to local (rack-1, tor-11/tor-12). Here is the sequence (from the perspective of tor-11) -

1. To start the VM is remote (45.0.3.29 on remote rack-2). Equivalent iproute2 command -
(tor-11) ip neigh add 45.0.3.29 dev vlan1003 lladdr 00:02:00:00:00:9c extern_learn nud noarp proto zebra

2. The VM then moves to rack-1 and tor-12 learns about it. tor-12 then advertises a type-2 MAC-IP route which is install by tor-11 as a local neigh. Equivalent iproute2 command -
(tor-11) ip neigh replace 45.0.3.29 dev vlan1003 lladdr 00:02:00:00:00:9c mh_peer_sync nud stale proto zebra **state_override_weak**

This neigh replace will fail because of the "state_override_weak" flag. To fix this, very basic MM problem, we have dropped use of that flag.

@pjdruddy does that answer your question?

Reply:
yeah, got it.